core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -380,13 +386,19 @@ private BasicInvocationStage handleNonTxWriteCommand(InvocationContext ctx, Writ
          }
          if (!(ce instanceof OutdatedTopologyException) && !(ce instanceof SuspectException))
             throw t;
+         WriteCommand writeCommand = (WriteCommand) rCommand;
+         if (writeCommand.hasFlag(Flag.ZERO_LOCK_ACQUISITION_TIMEOUT)) {
+            // the command is expected to not wait
```

Comment:
I think that in Hibernate ORM 2LC on dist cache PFERs run remotely, too. Not sure if these are sync/async, but PFERs should not imply CACHE_MODE_LOCAL.
What's the purpose of zero lock timeout rather than fail-fast?